pyro/infer/mcmc/hmc.py

```diff
@@ -78,6 +87,9 @@ def __init__(self, model, step_size=None, trajectory_length=None,
             self.trajectory_length = 2 * math.pi  # from Stan
         self.num_steps = max(1, int(self.trajectory_length / self.step_size))
         self.adapt_step_size = adapt_step_size
+        self.mass_structure = mass_structure
+        # Don't need to adapt mass if the mass structure is identity.
+        self.adapt_mass = adapt_mass and self.mass_structure != 'identity'
```

Comment:
Can we just use the `mass_structure` param with `identity` as the default, since the `adapt_mass` is redundant? We already have a few parameters that are dependent, and we would like to reduce dependent parameters as much as possible.